modules/c++/nitf/include/nitf/CppImageSubheader.hpp

```diff
@@ -0,0 +1,683 @@
+    {
+        return mImageTitle;
+    }
+
+    inline void setImageTitle(const std::string& imageTitle)
+    {
+        mImageTitle = imageTitle;
+    }
+
+    inline std::string getImageSecurityClassification() const
+    {
+        return mImageSecurityClassification;
+    }
+
+    void setImageSecurityClassification(const std::string& classification);
+
+    inline std::string getClassificationSystem() const
+    {
+        return mClassificationSystem;
+    }
+
+    inline void setClassificationSystem(const std::string& classificationSystem)
+    {
+        mClassificationSystem = classificationSystem;
+    }
+
+    inline std::string getCodewords() const
+    {
+        return mCodewords;
+    }
+
+    inline void setCodewords(const std::string& codewords)
+    {
+        mCodewords = codewords;
+    }
+
+    inline std::string getControlAndHandling() const
+    {
+        return mControlAndHandling;
+    }
+
+    inline void setControlAndHandling(const std::string& controlAndHandling)
+    {
+        mControlAndHandling = controlAndHandling;
+    }
+
+    inline std::string getReleasingInstructions() const
+    {
+        return mReleasingInstructions;
+    }
+
+    inline void setReleasingInstructions(const std::string& instructions)
+    {
+        mReleasingInstructions = instructions;
+    }
+
+    void setDeclassificationType(const std::string& declassType);
+    inline std::string getDeclassificationType() const
+    {
+        return mDeclassificationType;
+    }
+
+    inline DateTime getDeclassificationDate() const
+    {
+        return mDeclassificationDate;
+    }
+
+    inline void setDeclassificationDate(const DateTime& date)
+    {
+        mDeclassificationDate = date;
+    }
+
+    void setDeclassificationExemption(const std::string& exemption);
+    inline std::string getDeclassificationExemption() const
+    {
+        return mDeclassificationExemption;
+    }
+
+    void setDowngrade(const std::string& downgrade);
+    inline std::string getDowngrade() const
+    {
+        return mDowngrade;
+    }
+
+    inline DateTime getDowngradeDateTime() const
+    {
+        return mDowngradeDateTime;
+    }
+
+    inline void setDowngradeDateTime(const DateTime& datetime)
+    {
+        mDowngradeDateTime = datetime;
+    }
+
+    inline std::string getClassificationText() const
+    {
+        return mClassificationText;
+    }
+
+    inline void setClassificationText(const std::string& text)
+    {
+        mClassificationText = text;
+    }
+
+    void setClassificationAuthorityType(const std::string& classificationType);
+    inline std::string getClassificationAuthorityType() const
+    {
+        return mClassificationAuthorityType;
+    }
+
+    inline std::string getClassificationAuthority() const
+    {
+        return mClassificationAuthority;
+    }
+
+    inline void setClassificationAuthority(const std::string& authority)
+    {
+        mClassificationAuthority = authority;
```

Comment:
I think you want all this security stuff to be in its own class because other subheaders will have the exact same security section